### PR TITLE
Fastparquet

### PR DESCRIPTION
**Pull request by joemarshall**
adds fastparquet and the cramjam backend.

In progress (both build on my machine, untested so far)

### CODE REVIEW

**Comment by joemarshall** [comment]
Weird, has an attack of builds on my machine-itis. Something wrong with the rust settings.

**Comment by ryanking13** [comment]
```
💥 maturin failed                                                               
  Caused by: Invalid PYO3_CONFIG_FILE                                           
  Caused by: missing value for ext_suffix
```

I think I also experienced the same build issue when building pyxel. Unfortunatly, I don't remember what the fix was. Perhaps our [`pyo3_config.ini`](https://github.com/pyodide/pyodide/blob/main/pyodide-build/pyodide_build/tools/pyo3_config.ini) is not compatible with some maturin / pyo3 versions. IIRC adding `ext_suffix` value to the pyo3_config.ini didn't help, but probably you can try that.

@massense Do you have any ideas about this error?

**Comment by messense** [comment]
ext_suffix requires updating to pyo3 0.18.1.

See https://github.com/PyO3/pyo3/pull/2926

**Comment by messense** [comment]
Hopefully https://github.com/PyO3/maturin/pull/1491 will make in tree build works out of the box with the current `pyo3-config.ini` setup.

**Comment by ryanking13** [comment]
Wow, thanks a lot @messense!

**Comment by joemarshall** [comment]
I just found this. Thanks ever so much @messense - once the maturin version is on a pypi release this PR should work.

**Comment by messense** [comment]
[maturin v0.14.14](https://github.com/PyO3/maturin/releases/tag/v0.14.14) is out, please give it a try.

**Comment by joemarshall** [comment]
@ryanking13 this PR makes a couple of minor changes to the build - and updates the pyo3 config to use the correct python version. I don't mind splitting them out if that is neater, but otherwise I think it is pretty close to ready. 

I added a note to the package building docs about maturin - it works fine as long as the version is >=0.14.14. I patch that into fastparquet here and it works fine.

**Comment by joemarshall** [comment]
Oh and it also adds cramjam, because that's a dep of fastparquet by the way. Useful little compression library with pretty much every possible compression algorithm.

**Review by ryanking13 — APPROVED** [review]
Thanks a lot for the investigation @joemarshall. It is sad that there is no good global pyo3_config.ini that works for every maturin version, but it is worth it that we now know how to handle it :) and patching pyproject.toml or providing custom PYO3_CONFIG_FILE is not a big deal.

It seems like cramjam tests are failing because of numpy, otherwise looks good to me.

File: packages/fastparquet/test_fastparquet.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import pytest
+from pytest_pyodide import run_in_pyodide
+
+COMPRESSIONS = ("SNAPPY", "GZIP", "LZ4", "BROTLI", "ZSTD")
+
+
+# just check that we can read and write random data
+@pytest.mark.parametrize("compression", COMPRESSIONS)
+@run_in_pyodide(packages=["fastparquet", "packaging"])
```
[the remark this answers is not in the document]
Comment (joemarshall) [reply packages/fastparquet/test_fastparquet.py:9]: Oops, was because I missed a dependency in fastparquet, fixed now.

**Comment by joemarshall** [comment]
@ryanking13 Removed the numpy dependency in tests (and fixed the packaging weirdness). Think it should be good to go now.

**Comment by ryanking13** [comment]
Thanks again @joemarshall!

**Comment by rth** [comment]
The test  `test_simple_table[safari-BROTLI]` seems to be failing with Safari also on master. So at least we should maybe xfail it.

**Pull request merged**